### ISSUE TITLE
Regarding the losses being monitored per epoch and LR

### ISSUE THREAD

**Issue by vizer1993** [fizyr/keras-retinanet#529]
The per epoch losses (total loss, regression loss and classification loss) being monitored - are they that of the training data or validation (if validation split exists - I haven't found an argument yet in train.py that lets me split my training dataset into train and validation sets). Also, can accuracies be monitored as well along with the losses. 

Another thing I found is that train.py never seems to converge; it does however invoke the ReduceLROnPlateau argument when the loss plateaus out (obviously) and then keeps training where-after the loss then further decreases by a small fraction. Should I just stop training any further when it does that or let it continue? 

Snippet of how my training looks like:

> Epoch 16/150 2965/2965 [==============================] - 22779s 8s/step - loss: 3.9569e-04 - regression_loss: 3.9504e-04 - classification_loss: 6.4717e-07

> Epoch 0016: saving model to  _save_path_

> Epoch 0016: ReduceLROnPlateau reducing learning rate to 9.99999997475e-08.

> Epoch 17/150 1568/2965 [===============> . . . . . . . . . . . . . . ] - ETA: 2:58:45 - loss: 3.8182e-04 - regression_loss: 3.8118e-04 - classification_loss: 6.2010e-07

**Dataset:** 283 labelled images; Number of Classes: 1; multiple labels (>= 100) per image ≡ 29650 training samples (rows) when put in csv format.
**Training Info:** using ResNet50, training initialized with best weights obtained from a previous training session with batch size 1, 10k steps.
**Current training session arguments:** batch_size = 10, steps = 2965, on Tesla P40.

**Comment by hgaiser** [comment]
The losses being monitored are that of the training set. You can pass a validation set using `--val-annotations` for the CSV dataset. It will compute mAP accuracies on your val annotations.

It could be that your mAP doesn't change after a while, while the loss still changes. The network isn't learning to detect more objects then, but it becomes more confident about its detections (hence the loss decreases).

Closing this as it is not an issue. Please continue further discussions on Slack (see README).

**Comment by vizer1993** [comment]
But in that case, the validation set has to be separated out from the training dataset, right i.e., separate csv files for training and validation?

**Comment by hgaiser** [comment]
Yeah exactly. Keras-retinanet doesn't do that for you though, you have to
take care of that yourself.

**Comment by vizer1993** [comment]
Got it. Thanks.

**Comment by juanmanuellopezcorrea** [comment]
Hi Vizer 1993
How do you do to save the model in each epoch?

**Comment by vizer1993** [comment]
> Hi Vizer 1993
> How do you do to save the model in each epoch?

use the ModelCheckpoint argument from Keras' callbacks API: [Keras Callbacks API](https://keras.io/api/callbacks/)]
